tests/sidecar_test.go

```diff
@@ -0,0 +1,405 @@
+//
+// DISCLAIMER
+//
+// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Kaveh Vahedipour <kaveh@arangodb.com>
+//
+package tests
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	driver "github.com/arangodb/go-driver"
+	api "github.com/arangodb/kube-arangodb/pkg/apis/deployment/v1alpha"
+	kubeArangoClient "github.com/arangodb/kube-arangodb/pkg/client"
+	"github.com/arangodb/kube-arangodb/pkg/util"
+	"github.com/dchest/uniuri"
+	v1 "k8s.io/api/core/v1"
+)
+
+type sideCarTest struct {
+	shortTest bool
+	name      string
+	mode      api.DeploymentMode
+	version   string
+	image     string
+	imageTag  string
+	sideCars  map[string][]v1.Container
+}
+
+type SideCarTest interface {
+	IsShortTest() bool
+	Mode() api.DeploymentMode
+	Name() string
+	Image() string
+	Version() driver.Version
+	GroupSideCars(string) []v1.Container
+	AddSideCar(string, v1.Container)
+	ClearGroupSideCars(group string)
+}
+
+func (s *sideCarTest) IsShortTest() bool {
+	return s.shortTest
+}
+func (s *sideCarTest) Name() string {
+	return s.name
+}
+func (s *sideCarTest) Mode() api.DeploymentMode {
+	return s.mode
+}
+func (s *sideCarTest) Version() driver.Version {
+	return driver.Version(s.version)
+}
+func (s *sideCarTest) GroupSideCars(group string) []v1.Container {
+	if s.sideCars == nil {
+		s.sideCars = make(map[string][]v1.Container)
+	}
+	return s.sideCars[group]
+}
+
+func (s *sideCarTest) AddSideCar(group string, container v1.Container) {
+	if s.sideCars == nil {
+		s.sideCars = make(map[string][]v1.Container)
+	}
+	s.sideCars[group] = append(s.sideCars[group], container)
+}
+
+func (s *sideCarTest) Image() string {
+	imageName := "arangodb/arangodb"
+	if s.image != "" {
+		imageName = s.image
+	}
+	imageTag := "latest"
+	if s.imageTag != "" {
+		imageTag = s.imageTag
+	}
+	return fmt.Sprintf("%s:%s", imageName, imageTag)
+}
+func (s *sideCarTest) ClearGroupSideCars(group string) {
+	s.sideCars[group] = nil
+}
+
+// TestSideCars tests side car functionality
+func TestSideCars(t *testing.T) {
+	runSideCarTest(t, &sideCarTest{
+		version: "3.4.7",
```

Comment:
Does the test always use `arangodb/arangodb:3.4.7`?

Reply:
until changed yes